src/databases/block_database.cpp

```diff
@@ -455,6 +455,8 @@ bool block_database::unindex(const hash_digest& hash, size_t height,
     if (!element)
         return false;
 
+    BITCOIN_ASSERT(hash == element.key());
```

Comment:
In non-debug builds hash is an unused parameter. Therefore the parameter name should be wrapped in the DEBUG_ONLY(hash) macro.